### ISSUE TITLE
Reconsider behaviour for `GetRangeByHeight`

### ISSUE THREAD

**Issue by renaynay** [celestiaorg/celestia-node#1163]

**Comment by renaynay** [comment]
![Screenshot 2022-09-26 at 12 12 56](https://user-images.githubusercontent.com/41963722/192251469-ffaa1aad-c8fa-4d74-a1bc-1f12d12d8a52.png)
from https://github.com/celestiaorg/celestia-node/pull/1135

**Comment by walldiss** [comment]
Should we actually return an `ErrNotFound` in situation where partial data is returned? I think it might lead to confusion with situation when there is no data at all. Also usually in api design if error is return, response data is not considered, but only error is handled. Maybe add another statusCodePartial to keep status explicit in situations when partial data is returned?

**Comment by Wondertan** [comment]
I agree with returning ErrNotFound instead of partial data, for the reason of partial data response being a very rare case that will complicate the implementation for very minimal gain. Currently, all peers have all the headers, in future with pruning, peers will have either peers with all headers or peers with headers for unboding window(3 weeks since head). If this changes it some point, we can reconsider making partial responses.
